mountpoint-s3-crt/src/s3/client.rs

```diff
@@ -61,6 +61,55 @@ pub struct ClientConfig {
 
     /// The region
     region: Option<String>,
+
+    /// List of network interfaces to be used by the client.
+    ///
+    /// The client will determine the logic for balancing requests over the network interfaces
+    /// according to its own logic.
+    network_interface_names: Option<NetworkInterfaceNames>,
+}
+
+/// This struct bundles together the list of owned strings for the network interfaces, and the
+/// cursors pointing to them.
+///
+/// This is necessary because the CRT expects a pointer to the list of cursors,
+/// and the lifetime of the cursors must meet or exceed the lifetime of the [ClientConfig].
+/// By wrapping the array of cursors in this struct, we know the cursors will be valid
+/// and we can ensure the memory will be freed when dropped at the same time as [ClientConfig].
```

Comment:
I'm pretty sure the CRT [copies these strings out](https://github.com/awslabs/aws-c-s3/blob/0ab4d58ef0bd97970d43828cb6b57a3de5747343/source/s3_client.c#L597-L598), so we don't need machinery to keep them around like this forever, only until `aws_s3_client_new` is finished. That means you can probably just create the `Vec<aws_byte_cursor>` inside the `Client::new` method and throw it away once you're done.